src/test/java/com/google/jenkins/plugins/computeengine/integration/CleanLostNodesWorkIT.java

```diff
@@ -0,0 +1,210 @@
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.PrefixedOutputStream;
+import org.jvnet.hudson.test.RealJenkinsRule;
+import org.jvnet.hudson.test.TailLog;
+
+public class CleanLostNodesWorkIT {
+    private static final Logger log = Logger.getLogger(CleanLostNodesWorkIT.class.getName());
+    private static final String LOG_RECORDER_NAME = "CleanLostNodesWork log recorder";
+    /*
+    Provisioning a Cloud VM encounters two primary delays:
+    1. Initiation Delay: The plugin's initiation of the request is slow and requires optimization.
+    2. Creation Time: The VM undergoes creation and must reach a 'running' state.
+
+    During these phases, the VM's timestamp label retains its initial request value.
+    This can lead to scenarios where, before the current controller's CleanLostNodesWork can update the label,
+    another controller may mistakenly identify and delete the VM as an orphan.
+
+    To solve this race condition from happening, we need to set sufficiently higher value for the recurrence period,
+    which prevents the race condition as well as doesn't make the tests too slow (which are already slow).
+
+    Based on tests conducted with intervals of 30s, 45s, and 60s, a 60s delay currently proves most effective.
+    */
+    private static final long CLEAN_LOST_NODES_WORK_RECURRENCE_PERIOD = 60 * 1000;
+    private static final Map<String, String> googleLabels = getLabel(CleanLostNodesWorkIT.class);
+
+    @Rule
+    public RealJenkinsRule rj1 = new RealJenkinsRule();
+
+    @Rule
+    public RealJenkinsRule rj2 = new RealJenkinsRule();
+
+    @Before
+    public void init() throws Throwable {
+        Stack<PrefixedOutputStream.Color> colors = new Stack<>() {
+            {
+                push(PrefixedOutputStream.Color.BLUE);
+                push(PrefixedOutputStream.Color.RED);
+            }
+        };
+        for (var rj : List.of(rj1, rj2)) {
+            rj.javaOptions("-Djenkins.cloud.gcp.cleanLostNodesWork.recurrencePeriod="
+                            + CLEAN_LOST_NODES_WORK_RECURRENCE_PERIOD)
+                    .withLogger(CleanLostNodesWork.class, Level.FINEST)
+                    .withColor(colors.pop());
+            rj.startJenkins();
+            rj.runRemotely(r -> {
+                initCredentials(r);
+                var cloud = initCloud(r);
+                var instanceConfig = instanceConfigurationBuilder()
+                        .numExecutorsStr(NUM_EXECUTORS)
+                        .labels(LABEL)
+                        .oneShot(false)
```

Comment:
Why not
```suggestion
                        .oneShot(true)
```
? This should be considered the normal mode of operation of the plugin AFAIK.

Reply:
right, my mistake. thank you